lib/linux.js

```diff
@@ -83,83 +87,65 @@ function setAvailableApps(cb) {
 	// `which` all commands and expect stdout to return a positive
 	const whichCmd = `which -a ${names.join('; which -a ')}`;
 
-	childProcess.exec(whichCmd, (err, stdout) => {
-		if (!stdout) {
-			throw new Error('None of the apps were found');
-		}
+	return exec(whichCmd)
+		.then(function (stdout) {
+			if (!stdout) {
+				throw new Error('None of the apps were found');
+			}
 
-		// it may return aliases so we only want the real path
-		// and only the executable name. i.e. 'foo' from /bin/foo
-		stdout = stdout.trim().split('\n');
+			// it may return aliases so we only want the real path
+			// and only the executable name. i.e. 'foo' from /bin/foo
+			stdout = stdout.trim().split('\n');
 
-		stdout.forEach(el => {
-			// it's an alias
-			if (el[0] !== path.sep) {
-				return;
-			}
+			stdout.forEach(el => {
+				// it's an alias
+				if (el[0] !== path.sep) {
+					return;
+				}
 
-			el = el.split(path.sep).pop();
+				el = el.split(path.sep).pop();
 
-			availableApps.push(availableAppsDict[el]);
+				availableApps.push(availableAppsDict[el]);
+			});
 		});
-
-		cb(availableApps);
-	});
 }
 
-exports.get = function get(cb) {
+exports.get = function get() {
 	if (!availableApps) {
-		return setAvailableApps(() => {
-			get(cb);
-		});
+		return setAvailableApps().then(get);
 	}
 
-	cb = cb || function () {};
-
-	let found = false;
-
-	availableApps.forEach(app => {
-		if (!app.get || found) {
-			return;
-		}
-
-		childProcess.execFile(app.cmd, app.get, (err, stdout) => {
-			if (err || !stdout || found) {
-				return;
-			}
-
-			found = true;
+	const app = availableApps.find(app => {
+		return app.get !== undefined;
+	});
 
+	return execFile(app.cmd, app.get)
+		.then(stdout => {
 			if (typeof app.transform === 'function') {
-				stdout = app.transform(stdout);
+				return app.transform(stdout);
 			}
 
-			cb(null, stdout);
+			return stdout;
 		});
-	});
 };
```

Comment:
Not sure who implemented that originally, but maybe @sindresorhus knows why the check was added in the original code.